xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/xwikivars.vm

```diff
@@ -21,7 +21,7 @@
 ##!unique=request
 #template("frequentlyUsedDocs.vm")
 #set ($isGuest = $xcontext.getUser().equals('XWiki.XWikiGuest'))
-#set ($isSuperAdmin = ($xcontext.user == 'XWiki.superadmin'))
+#set ($isSuperAdmin = ($xcontext.user == 'XWiki.superadmin') || ($xcontext.user == "${services.wiki.mainWikiId}:XWiki.superadmin"))
```

Comment:
Feels like we should stop have any kind of logic in this template and instead call some API which check that as it's supposed to be checked.

Reply:
I applied you proposal in the last commit